distribution/docker/Dockerfile

```diff
@@ -0,0 +1,55 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+FROM maven:3-jdk-8 as builder
+
+COPY . /src
+WORKDIR /src
+RUN mvn install -ff -DskipTests -Dforbiddenapis.skip=true -Pdist -Pbundle-contrib-exts
+
+RUN \
+ VER=$(mvn -B org.apache.maven.plugins:maven-help-plugin:3.1.1:evaluate -Dexpression=project.version -q -DforceStdout=true -f pom.xml 2>/dev/null) \
+ && tar -zxf ./distribution/target/apache-druid-${VER}-bin.tar.gz -C /opt \
+ && ln -s /opt/apache-druid-${VER} /opt/druid
+
+RUN wget -O /opt/druid/extensions/mysql-metadata-storage/mysql-connector-java-5.1.38.jar http://central.maven.org/maven2/mysql/mysql-connector-java/5.1.38/mysql-connector-java-5.1.38.jar \
```

Comment:
For lookups to be able to use MySQL, currently this JAR needs to also be in /opt/druid/lib

Sorry for this not being obvious, there's a related issue:

https://github.com/apache/incubator-druid/issues/3770

Reply:
IMHO that should not be a blocker for deployment since it is an extension and even a conditional use case for the extension.

If people are needing to get to that level of detail in what is and is not in the docker image I would prefer they either build new layers on top of this one, or use this one as a template and make whatever modifications they need. AKA: keep the scope here for "get an experimental cluster up as quickly as possible"

Reply:
Hmm, I'd agree if the image just included core Druid. 

However it has explicitly included all extensions both core and contributed so I think it's worth putting our dependencies in the right place to ensure those extensions work as advertised.

Lest someone give up on their experimental cluster because they're unable to track down why they're getting strange classpath errors.

Reply:
ok so i can soft-link it?